proposals/shared-everything-threads/Overview.md

```diff
@@ -498,29 +502,32 @@ behave mostly like [sealed] JS objects, and their identities survive round trips
 across thread boundaries. Additionally, to facilitate calling methods on these shared objects, they
 have [realm-local prototypes][tls-prototypes] just like shared structs are proposed to have in JS.
 
-The shared versions of the JS API types (besides memories) will be represented with new
-constructors:
+The shared versions of the JS API types will share constructors with their existing unshared
+versions, but will take an additional `shared: bool` option in an options bag argument. These
+constructors will be extended to take this new option:
 
- - `WebAssembly.SharedFunction`
- - `WebAssembly.SharedTable`
- - `WebAssembly.SharedGlobal`
- - `WebAssembly.SharedTag`
- - `WebAssembly.SharedException`
+ - `WebAssembly.Function`
+ - `WebAssembly.Table`
+ - `WebAssembly.Global`
+ - `WebAssembly.Tag`
+ - `WebAssembly.Exception`
 
-These new types provide an API that mirrors that provided by their unshared analogues. Like shared
-memories, shared tables must be initialized with a maximum size. Shared exceptions may only be
-created with shared tags. Storing unshared data into a shared global or table throws an exception.
+These new shared objects provide an API that mirrors that provided by their unshared analogues. Like
+shared memories, shared tables must be initialized with a maximum size. Shared exceptions may only
+be created with shared tags. Storing unshared data into a shared global or table throws an
+exception. In addition, all of these objects will be extended with a read-only `shared` boolean
+property that can be used to test whether the object is shared or not.
```

Comment:
Same here, it should be part of the type descriptor returned by the `type` methods.

Reply:
FWIW shared `WebAssembly.Memory` already have this read-only field mechanism.

Reply:
Well, I suppose there is no harm in keeping that around as a redundant legacy feature.